hypertrace-core-graphql-span-schema/src/main/java/org/hypertrace/core/graphql/span/dao/SpanLogEventResponseConverter.java

```diff
@@ -54,18 +54,37 @@
       LogEventsResponse logEventsResponse) {
     return Observable.fromIterable(logEventsResponse.getLogEventsList())
         .concatMapSingle(
-            logEventsResponseVar -> this.convert(attributeRequests, logEventsResponseVar))
-        .collect(Collectors.groupingBy(logEvent -> (String) logEvent.attribute(foreignIdAttribute)))
+            logEventsResponseVar ->
```

Comment:
What's the motivation for the change here? Since we were assuming the attribute existed in the response map and now we're guaranteeing it, the old logic would have also worked, right?

Reply:
If the original for logs didn't have the spanId attribute we shouldn't return it back either. So, this part needed changes to filter out the spanId returned if it wasn't requested.

Reply:
based on the way the response object is designed, it's safe to return extra data (because it's not a programmatic access but instead via the schema, we already know at request time what values of the map will be read by the serializer - that's how we generated the requested attributes)

Reply:
yeah right spanId filtering would be automatically done, the problem was building the mapping of `spanId -> List<LogEvent>`. Currently the code was first converting LogEvent (gateway) -> LogEvent (gql) and then mapping spanId to logEvents, but now it is possible that spanId attribute would be lost after conversion.

Reply:
got it - because the attribute map conversion is based on the requested rather than returned data, and we don't save the extra attribute into the request. makes sense.